mathesar_ui/src/api/rpc/data_modeling.ts

```diff
@@ -61,7 +61,17 @@ export const data_modeling = {
       database_id: number;
       table_oid: number;
     },
-    Record<string, string>
+    Record<
+      string,
+      {
+        type: string;
+        details?: {
+          mathesar_casting?: boolean;
+          group_sep: string;
+          decimal_p: string;
+        };
+      }
+    >
```

Comment:
After a call with @mathemancer, it looks like the properties `group_sep`, and `decimal_p` are optional when `details` is present. The current type definition makes them required.

Since the casting related properties are not required directly by the frontend and is only stored and passed around for other requests, I'd suggest:

```suggestion
    Record<
      string,
      {
        type: string;
        details?: Record<string, unknown>;
      }
    >
```